test/test-Baselines-benchmark.js

```diff
@@ -0,0 +1,103 @@
+/* globals require, exports, XPCOMUtils, Task, Services, ClientID */
+"use strict";
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/ClientID.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+const {Benchmark, bLog} = require("./lib/Benchmark.js");
+const {getTestActivityStream} = require("./lib/utils");
+
+let app;
+let ACTIVITY_STREAMS_URL;
+
+const openAndCloseActivityStreamTab = Task.async(function*(bench, openUrl) {
+  let tabData = {};
+
+  // Start the benchmark timer here
+  bench.startTimer();
+  const promiseOnTabShow = new Promise(resolve => {
+    tabs.open({
+      url: openUrl,
+      onPageShow: tab => {
+        tabData.tab = tab;
+        resolve();
+      }
+    });
+  });
+
+  const promiseOnPerfLogComplete = new Promise(resolve => {
+    function onPerfLogComplete(subject, topic, data) {
+      Services.obs.removeObserver(onPerfLogComplete, "performance-log-complete");
+      resolve();
+    }
+    Services.obs.addObserver(onPerfLogComplete, "performance-log-complete", false);
+  });
+
+  const promiseSessionLogComplete = new Promise(resolve => {
+    function onNotify(subject, topic, data) {
+      Services.obs.removeObserver(onNotify, "tab-session-complete");
+      tabData.notifyData = JSON.parse(data);
+      resolve();
+    }
+    Services.obs.addObserver(onNotify, "tab-session-complete", false);
+  });
+
+  yield Promise.all([promiseOnTabShow, promiseOnPerfLogComplete]);
+  // Since we're only interested in the performance-log-complete message, manually stop the timer here
+  bench.stopTimer();
+
+  yield new Promise(resolve => {
+    tabData.tab.close(resolve);
+  });
+  yield promiseSessionLogComplete;
+});
+
+/*
+ * Creates a benchmark object for message performance_log_complete
+ *
+ * Params:
+ *   benchTime is an integer in microseconds
+ * Returns:
+ *   a benchmark object
+ */
```

Comment:
I'd suggest using jsdoc style for this, as there are lots of good (IDE and standalone) tools for auto-generating and doing interesting stuff with such comments.  Github even (mostly) knows how to highlight them:

```js
/**
 * Creates a benchmark object for message performance_log_complete
 *
 * @param {benchTime}     an integer in microseconds
 * @returns {Benchmark} 
 */
```

https://github.com/mozilla/loop/blob/master/shared/js/utils.js#L389 has a bunch of examples, and http://usejsdoc.org/ is the canonical reference.